Sources/UIGestureRecognizer.swift

```diff
@@ -29,17 +29,19 @@ open class UIGestureRecognizer {
 
     public var state: UIGestureRecognizerState = .possible {
         didSet {
-            if state == oldValue { return }
+            if
+                state == oldValue || oldValue == .cancelled && state != .ended
+            {
+                return
+            }
+
             onStateChanged?()
+
             switch state {
-            case .failed, .cancelled:
-                // touchesCancelled(touches: Set<UITouch>, with: UIEvent)
+            case .recognized, .ended, .failed:
                 state = .possible
```

Comment:
ok so if we `failed` then we call `onStateChanged` which presumably is aware of that current state, and immediately after that we set the state back to `.possible`?

when does `touchesCancelled` get called?

Reply:
this is basically the same as before, we just don't immediately reset the state to `.possible` when cancelled. As far as I can see `touchesCancelled` only gets called from `cancelOtherGestureRecognizersThatShouldNotRecognizeSimultaneously()`

Reply:
i guess this makes sense. would love to better understand what ios does here at some point but doesn't have to be in this PR if it suits our use cases and doesn't do anything too crazy